fpdf/line_break.py

```diff
@@ -9,47 +17,144 @@
 
 
 class Fragment:
-    def __init__(self, style: str, underlined: bool, characters: str = None):
-        self.style = style
-        self.underline = underlined
-        self.characters = [] if characters is None else characters
+    """
+    A fragment of text with font/size/style and other associated information.
+    """
+
+    def __init__(self, characters: Union[list, str], graphics_state: dict, k: float):
```

Comment:
The properties in this `graphics_state` are know "in advance",
so maybe a `typing.NamedTuple` subclass would be better than a `dict` here,
in order to make it clearer what properties this `graphics_state` are expected to contain ?

Also I'm not sure about the naming...
This is very specific to text rendering, and doesn't have much to do with the PDF concept of "Graphics State", right?
It reminds me a bit of `fpdf.TitleStyle` actually.
Maybe `text_style`?

Reply:
This is simply a (shallow) copy of the current graphics state from `graphics_state.GraphicsStateMixin()`. Converting it into a different type here doesn't really make much sense.

The name is taken directly from the PDF specs (Section 8.4: Graphics State), which it is modelled after.

Reply:
Ok then